desc/objectives/_profiles.py

```diff
@@ -10,35 +10,214 @@
 
 from .normalization import compute_scaling_factors
 from .objective_funs import _Objective
+from .utils import _parse_callable_target_bounds
+
+
+class Pressure(_Objective):
+    """Target pressure profile.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium that will be optimized to satisfy the Objective.
+    target : {float, ndarray, callable}, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        Must be broadcastable to Objective.dim_f. If a callable, should take a
+        single argument `rho` and return the desired value of the profile at those
+        locations.
+    bounds : tuple of {float, ndarray, callable}, optional
+        Lower and upper bounds on the objective. Overrides target.
+        Both bounds must be broadcastable to to Objective.dim_f
+        If a callable, each should take a single argument `rho` and return the
+        desired bound (lower or upper) of the profile at those locations.
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _scalar = False
+    _linear = False  # in general, ie for kinetic equilibria
```

Comment:
Should we overwrite this to be linear if the objective is initialized/built with an equilibrium that has a pressure profile assigned? 

Technically a user could build with a pressure equilibrium and actually optimize with a kinetic one, in which case my suggestion would break things. But that is also a rare use case.

Reply:
Probably not worth it, in that case using `FixPressure` would be easier. The idea with this was to be able to have bounds etc in which case its not linear anyways.